### PR TITLE
Add a sample app and systemd units

### LINKED ISSUE #3: Add  example systemd units to docs
E.g.:

```ini
# warp.service
[Unit]
Description=Warp
[Service]
Type=notify
WatchdogSec=5
ExecStart=path-to-my-haskell-binary
```

```ini
# warp.socket
[Unit]
Description=Warp
[Socket]
ListenStream=80
[Install]
WantedBy=sockets.target
```

```
systemctl enable --now warp.socket
```

### PR DESCRIPTION
**Pull request by sandydoo**
Resolves #3.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,106 @@
...
+Let's configure your app to run as a socket-activated systemd service. We'll be using the sample app in [example/Main.hs](example/Main.hs).
+
+1. Create a new service unit at `/etc/systemd/system/warp.service` that'll run your Haskell binary:
+
+   ```systemd
+   [Unit]
+   Description=Warp
+   [Service]
+   Type=notify
+   WatchdogSec=5
```
Comment (roberth) [inline README.md:46]: I usually add margin, or does systemd decide that? Otherwise having the same number of seconds seems like cutting it really close.

**Comment by roberth** [comment]
I'm adding some tests in #8

File: README.md (hunk 1 above)
Comment (sandydoo) [reply README.md:46]: Good point. Maybe 30 seconds?

Comment (roberth) [reply README.md:46]: Sgtm

**Comment by sandydoo** [comment]
@roberth, should we close this in favour of #8? The nixos tests are a nice touch 🙂

**Comment by roberth** [comment]
It'd be cleaner to rebase that one. It's just your code plus a bit of mine. And then a shitty formatting commit which really doesn't matter but makes the diff stupid. Sorry about that.

**Review by roberth — APPROVED** [review]

**Pull request merged**